src/main/java/DBConnector.java

```diff
@@ -0,0 +1,179 @@
+
+    public static void createDatabase(String dbname) {
+        dbName = dbname;
+        url = "jdbc:sqlite:" + dbName;
+        dataSource.setUrl(url);
+        try (Connection con = dataSource.getConnection()) {
+            try (Statement statement = con.createStatement()) {
+            } catch (SQLException e) {
+                e.printStackTrace();
+            }
+        } catch (SQLException e) {
+            e.printStackTrace();
+        }
+    }
+
+    public static void createTable(String name) {
+        cardTableName = name;
+        try (Connection conn = DriverManager.getConnection(url);
+             Statement stmt = conn.createStatement()) {
+            stmt.executeUpdate("CREATE TABLE IF NOT EXISTS card ("
+                    + "id INTEGER PRIMARY KEY,"
+                    + "number text NOT NULL,pin TEXT NOT NULL,"
+                    + "balance INTEGER DEFAULT 0)");
+        } catch (SQLException e) {
+            System.out.println(e.getMessage());
+        }
+    }
+
+    public static void insertCard(Account account) {
+        String sql = "INSERT INTO " + cardTableName + " VALUES "
+                + "	(" + account.getId() + "," + account.getCardNumber() + "," + account.getPIN() + "," + account.getBalance() + ");";
+        try (Connection conn = DriverManager.getConnection(url);
+             Statement stmt = conn.createStatement()) {
+            int i = stmt.executeUpdate(sql);
+        } catch (SQLException e) {
+            System.out.println(e.getMessage());
+        }
+    }
+
+    public static int getCount() {
+        int id = 0;
+        try (Connection conn = DriverManager.getConnection(url);
+             Statement stmt = conn.createStatement()) {
+
+            ResultSet res = stmt.executeQuery("Select * from " + cardTableName + ";");
+            while (res.next()) {
+                id = res.getInt("id");
+            }
+
+        } catch (SQLException e) {
+            System.out.println(e.getMessage());
+        }
+        return id;
+    }
+
+    public static boolean containCard(String cardNumber) {
+        String contains = "SELECT count(number) FROM " + cardTableName + " WHERE number = '" + cardNumber + "';";
+        ResultSet res;
+        try (Connection con = DriverManager.getConnection(url)) {
+            try (Statement stmt = con.createStatement()) {
+                res = stmt.executeQuery(contains);
+                if (res.next()) {
+                    return res.getBoolean(1);
+                }
+            } catch (SQLException e) {
+                e.printStackTrace();
+            }
+        } catch (SQLException e) {
+            e.printStackTrace();
+        }
+        return false;
+    }
+
+    public static int login(String cardNumber, int PIN) {
+        String login = "SELECT id FROM " + cardTableName + " WHERE number = '" + cardNumber + "' AND pin = " + PIN + ";";
+        ResultSet res;
+        try (Connection con = DriverManager.getConnection(url)) {
```

Comment:
You can get a connection one time and use it, please think about it.

Reply:
check new code pls